src/entries/popup/pages/hw/gridplus/addressChoice.tsx

```diff
@@ -0,0 +1,160 @@
+import { motion } from 'framer-motion';
+import { fetchAddresses } from 'gridplus-sdk';
+import { FormEvent, useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Address } from 'wagmi';
+
+import { i18n } from '~/core/languages';
+import { SessionStorage } from '~/core/storage';
+import { truncateAddress } from '~/core/utils/address';
+import { Box, Button, Text } from '~/design-system';
+import { Checkbox } from '~/entries/popup/components/Checkbox/Checkbox';
+import { Link } from '~/entries/popup/components/Link/Link';
+import { Spinner } from '~/entries/popup/components/Spinner/Spinner';
+import { HARDWARE_WALLETS } from '~/entries/popup/handlers/walletVariables';
+import { useAccounts } from '~/entries/popup/hooks/useAccounts';
+import { ROUTES } from '~/entries/popup/urls';
+
+export type AddressesData = {
+  addresses: Address[];
+};
+
+export const AddressChoice = () => {
+  const navigate = useNavigate();
+  const { state } = useLocation();
+  const { sortedAccounts } = useAccounts();
+  const [formData, setFormData] = useState({
+    selectedAddresses: [] as string[],
+  });
+  const [loadingAddresses, setLoadingAddresses] = useState(true);
+  const [addresses, setAddresses] = useState<AddressesData['addresses']>([]);
+  const disabled = loadingAddresses || formData.selectedAddresses.length === 0;
+  const toggleAddress = (address: string) => {
+    const selected = formData.selectedAddresses.includes(address);
+    if (selected)
+      return setFormData({
+        selectedAddresses: formData.selectedAddresses.filter(
+          (currentAddress) => currentAddress !== address,
+        ),
+      });
+    return setFormData({
+      selectedAddresses: [...formData.selectedAddresses, address],
+    });
+  };
+  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const accountsToImport = formData.selectedAddresses.map((address, i) => ({
+      address,
+      index: i,
+    }));
+    navigate(ROUTES.HW_WALLET_LIST, {
+      state: {
+        accountsToImport,
+        deviceId: 'GridPlus',
+        accountsEnabled: accountsToImport.length,
+        vendor: 'GridPlus',
+        direction: state?.direction,
+        navbarIcon: state?.navbarIcon,
+      },
+    });
+  };
+  useEffect(() => {
+    const fetchWalletAddresses = async () => {
+      setLoadingAddresses(true);
+      let fetchedAddresses: Address[];
+      if (process.env.IS_TESTING === 'true') {
+        fetchedAddresses = HARDWARE_WALLETS.MOCK_ACCOUNT.accountsToImport.map(
+          (account) => account.address,
+        );
+      } else {
+        fetchedAddresses = (await fetchAddresses()) as Address[];
+      }
+      const nonExistingAddresses = fetchedAddresses
+        .map((address) => getAddress(address))
+        .filter(
+          (address) =>
+            !sortedAccounts.map((account) => account.address).includes(address),
+        );
```

Comment:
is the idea here to not let the user select accounts that he already have in the extension? 
lets say he had the pk in the extension AND in the grid+ lattice?
this should prevent him from importing that account from the gridplus?

Reply:
Yes, precisely. It resulted in a thrown error when importing the same address another time.